Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -143,6 +146,11 @@ internal struct BinaryDecoder: Decoder {
         }
         if fieldNumber != 0 {
             consumed = false
+
+            // Reached the end of the current group.
+            if fieldWireFormat == .endGroup, groupFieldNumber == fieldNumber {
```

Comment:
Use `&&` instead of comma when the second clause doesn't depend on a binding from the first clause.